addons/web_editor/models/ir_qweb.py

```diff
@@ -144,43 +145,43 @@ def attributes(self, cr, uid, field_name, record, options,
         else:
             return attrs
 
-    def from_html(self, cr, uid, model, field, element, context=None):
-        Model = self.pool[element.get('data-oe-model')]
-        M2O = self.pool[field.comodel_name]
+    @api.model
+    def from_html(self, model, field, element):
+        Model = self.env[element.get('data-oe-model')]
+        M2O = self.env[field.comodel_name]
         field_name = element.get('data-oe-field')
         id = int(element.get('data-oe-id'))
         many2one_id = int(element.get('data-oe-many2one-id'))
 
-        if many2one_id and M2O.exists(cr, uid, [many2one_id]):
+        if many2one_id and M2O.browse(many2one_id).exists():
             # save the new id of the many2one
-            Model.write(cr, uid, [id], {
+            Model.browse(id).write({
                 field_name: many2one_id
-            }, context=context)
+            })
 
         # not necessary, but might as well be explicit about it
         return None
 
 
-class Contact(orm.AbstractModel):
+class Contact(models.AbstractModel):
     _name = 'ir.qweb.field.contact'
     _inherit = ['ir.qweb.field.contact', 'ir.qweb.field.many2one']
 
-    def attributes(self, cr, uid, field_name, record, options,
-                   source_element, g_att, t_att, qweb_context,
-                   context=None):
-        attrs = super(Contact, self).attributes(
-            cr, uid, field_name, record, options, source_element, g_att, t_att,
-            qweb_context, context=context)
+    @api.model
+    def attributes(self, field_name, record, options,
+                   source_element, g_att, t_att, qweb_context):
+        attrs = super(Contact, self).attributes(field_name, record, options, source_element, g_att, t_att, qweb_context)
         if getattr(record, field_name):
             return itertools.chain(attrs, [('data-oe-contact-options', json.dumps(options))])
         else:
             return attrs
 
     # helper to call the rendering of contact field
-    def get_record_to_html(self, cr, uid, ids, options=None, context=None):
-        node = self.record_to_html(cr, uid, 'record', {
-            'record': self.pool['res.partner'].browse(cr, uid, ids[0], context=context)},
-            options=options, context=context)
+    @api.model
+    def get_record_to_html(self, ids, options=None):
+        node = self.record_to_html('record', {
+            'record': self.env['res.partner'].browse(ids[0])},
```

Comment:
I think this method is only called from JS with single ID why not making this method handling single ID(as it is used for m2o)